Models/models.js

```diff
@@ -1,18 +1,57 @@
 const db = require('../db/connection');
 const documentation = require('../endpoints.json');
 const fs = require('fs/promises');
+const format = require('pg-format');
+
 
 async function getTopicsModel() {
     const { rows: topics } = await db.query("SELECT slug, description FROM topics;")
     return topics;
 }
 
+async function patchArticleModel(articleID, incCount) {
+    const articleQuery = await db.query('SELECT * FROM articles WHERE article_id=$1;', [articleID])
+    if (!articleQuery.rowCount) {
+        throw ({ code: "NOARTICLE" });
+    }
+    if (incCount === undefined) {
+        throw ({ code: "BAD_R" });
+    }
+    const result = await db.query(
+        `
+        UPDATE articles
+        SET votes = votes + $2
+        WHERE article_id = $1
+        RETURNING *;
+      `,
+        [articleID, incCount]
+    )
```

Comment:
Your initial query to the database to get the article by ID is not needed for the purposes of error handling. We can handle the 404 error by looking at the response from the update query.